OpenRA.Mods.Common/Widgets/LabelWidget.cs

```diff
@@ -52,20 +81,49 @@ protected LabelWidget(LabelWidget other)
 			GetText = other.GetText;
 			GetColor = other.GetColor;
 			GetContrastColor = other.GetContrastColor;
+			linePixelSpacing = other.linePixelSpacing;
+			font = other.font;
 		}
 
-		public override void Draw()
+		public int2 MeasureText(string text)
 		{
-			SpriteFont font;
-			if (!Game.Renderer.Fonts.TryGetValue(Font, out font))
-				throw new ArgumentException("Requested font '{0}' was not found.".F(Font));
+			var textSize = font.Value.Measure(text, linePixelSpacing.Value);
+			if (LineVAlign != LineVAlign.Collapsed)
+				textSize += new int2(0, linePixelSpacing.Value);
+			return textSize;
+		}
 
+		public string WrapText(string text)
+		{
+			return WidgetUtils.WrapText(text, Bounds.Width, font.Value);
+		}
+
+		public string TruncateText(string text)
+		{
+			return WidgetUtils.TruncateText(text, Bounds.Width, font.Value);
+		}
+
+		public int2 AutoSize(string text)
```

Comment:
This method name isn't very descriptive.  Please change it to something like `SizeToText`.